src/inference/dev_api/openvino/runtime/plugin_config.hpp

```diff
@@ -0,0 +1,416 @@
+
+        return it->second;
+    }
+
+    template <typename T, PropertyMutability mutability>
+    void apply_rt_info_property(const ov::Property<T, mutability>& property, const ov::RTMap& rt_info) {
+        if (!is_set_by_user(property)) {
+            auto rt_info_val = rt_info.find(property.name());
+            if (rt_info_val != rt_info.end()) {
+                set_user_property({property(rt_info_val->second.template as<T>())}, OptionVisibility::RELEASE);
+            }
+        }
+    }
+
+    ov::AnyMap read_config_file(std::string_view filename, std::string_view target_device_name) const;
+    ov::AnyMap read_env() const;
+    static ov::Any read_env(const std::string& option_name, std::string_view prefix, const ConfigOptionBase* option);
+    void cleanup_unsupported(ov::AnyMap& config) const;
+
+    std::map<std::string, ConfigOptionBase*> m_options_map;
+    using OptionMapEntry = decltype(m_options_map)::value_type;
+
+    // List of properties explicitly set by user via Core::set_property() or Core::compile_model() or ov::Model's
+    // runtime info
+    ov::AnyMap m_user_properties;
+
+    std::string_view get_help_message(const std::string& name = "") const;
+    void print_help() const;
+
+    bool m_is_finalized = false;
+
+    inline static const std::string_view m_allowed_env_prefix = "OV_";
+};
+
+class OptionRegistrationHelper {
+public:
+    OptionRegistrationHelper(PluginConfig* config, std::string_view name, ConfigOptionBase* option) {
+        if (config)
+            config->register_option(std::string{name}, option);
+    }
+};
+
+template <typename T>
+struct TypedOption : public ConfigOptionBase {
+    TypedOption(const T& default_val, std::string_view prop_name, std::string_view desc)
+        : ConfigOptionBase(prop_name, desc),
+          value(default_val) {}
+    T value;
+};
+
+template <typename T, OptionVisibility visibility_>
+struct ConfigOption : public TypedOption<T> {
+    ConfigOption(PluginConfig* config,
+                 std::string_view name,
+                 std::string_view prop_name,
+                 const T& default_val,
+                 std::string_view desc,
+                 std::function<bool(T)> validator = nullptr)
+        : TypedOption<T>(default_val, prop_name, desc),
+          validator(validator) {
+        OptionRegistrationHelper option(config, name, this);
+    }
+    constexpr static const auto visibility = visibility_;
+
+    void set_any(const ov::Any& any) override {
+        if (validator)
+            OPENVINO_ASSERT(validator(any.as<T>()), "Invalid value: ", any.as<std::string>());
+        this->value = any.as<T>();
+    }
+
+    ov::Any get_any() const override {
+        return ov::Any(this->value);
+    }
+
+    bool is_valid_value(const ov::Any& val) const override {
+        try {
+            auto v = val.as<T>();
+            return validator ? validator(v) : true;
+        } catch (std::exception&) {
+            return false;
+        }
```

Comment:
```suggestion
        if (val.is<T>()){
            return validator ? validator(val.as<T>()) : true
        } else{
            return false;
        }
```
Catch exception should not be required.

Reply:
It should not, but it is. I mentioned previously that is() method doesn't guarantee that as() works or doesn't work.